src/d3a/models/strategy/external_strategies/load.py

```diff
@@ -181,26 +182,8 @@ def event_market_cycle(self):
         if not self.should_use_default_strategy:
             self._calculate_active_markets()
             self._update_energy_requirement_future_markets()
-            self._reset_event_tick_counter()
-            market_event_channel = f"{self.channel_prefix}/events/market"
-            market_info = self.next_market.info
-            if self.is_aggregator_controlled:
-                market_info.update(default_market_info)
-            market_info['device_info'] = self._device_info_dict
-            market_info["event"] = "market"
-            market_info["area_uuid"] = self.device.uuid
-            market_info['device_bill'] = self.device.stats.aggregated_stats["bills"] \
-                if "bills" in self.device.stats.aggregated_stats else None
-            market_info["last_market_maker_rate"] = \
-                get_current_market_maker_rate(self.area.current_market.time_slot) \
-                if self.area.current_market else None
-            market_info['last_market_stats'] = \
-                self.market_area.stats.get_price_stats_current_market()
-            self.redis.publish_json(market_event_channel, market_info)
-            if self.is_aggregator_controlled:
-                self.redis.aggregator.add_batch_market_event(self.device.uuid,
-                                                             market_info,
-                                                             self.area.global_objects)
+            self.external_tick_timer.reset_event_tick_counter()
```

Comment:
I particularly like refactoring the tick algorithm to a separate class (ExternalTickTimer). There are 2 improvements I see:
1. this class is not exactly a Timer in the way Python perceives the term (a thread that performs some actions at a definite time). I think it is better to be renamed to ExternalTickCounter or similar. Then you can rename the `reset_event_tick_counter()` to just `reset()`
2. I think that the access to this class can only live in the ExternalMixin, in order to not repeat the `reset_event_tick_counter` calls